public/js/qr/history.js

```
@@ -0,0 +1,95 @@
+                <p>Error loading links</p>
+                <p style="font-size: 14px; margin-top: 8px;">${error.message}</p>
+                <button class="btn btn-sm btn-secondary" onclick="window.QRGenerator.loadUserLinks()" style="margin-top: 12px;">
+                    <i class="fas fa-redo"></i> Retry
+                </button>
+            </div>
+        `;
+    }
+};
+
+// --- Truncate long URLs for display ---
+QRGenerator.truncateUrl = function (url, maxLength) {
```

ESLint

[error] 'QRGenerator' is not defined.

(no-undef)